### PR TITLE
#152 add global.properties

### PR DESCRIPTION
**Pull request by vaa25**
connect to #152 add default.properties

### CODE REVIEW

**Comment by roma-ilnitsky** [comment]
Why we use local.properties.TEMPLATE if we have default.properties ?

**Comment by vaa25** [comment]
@roma-ilnitsky 
>  There you can replace any property you need from default.properties or add new secret property

**Comment by roma-ilnitsky** [comment]
I think you have to remove from local.properties.TEMPLATE 
```
# AESCryptTokenService (template keys)
aes.init.vector.hex=000102030405060708090A0B0C0D0E0F
#The secret key must equal the size of the key.  For
#256-bit encryption, the binary secret key is 32 bytes.
#For 128-bit encryption, the binary secret key is 16 bytes.
aes.key.hex=000102030405060708090A0B0C0D0E0F
```

**Comment by vaa25** [comment]
Overall this is a secret data and must be in local.properties. But by default it can be in default.properties too.

**Comment by roma-ilnitsky** [comment]
* **passed**

**Comment by alex-anakin** [comment]
What is reason to save property for `AESCryptTokenService` in default.properties and in local.properties at the same time?
About security. Our project is open-sourced. It stored in public repo. Anybody can read local.properties.TEMPLATE
Using http.port=8080 in default.properties is not good practice. It is default port for many web applications. I understand that we can replace it value in local.properties, but...


I think doubling of properties will turn deploying of the project into a charade.

**Comment by vaa25** [comment]
@alex-anakin @roma-ilnitsky 
`what is reason to save property for AESCryptTokenService in default.properties and in local.properties at the same time?` 
Really, why? If it is necessary for test, then it must be in test-local.properties. Nevertheless we can use that key to maintain our project until we have the real secret key, that we shall place into local.properties in future.
`Using http.port=8080 in default.properties is not good practice`
As you say many projects has that default property. So why our have to not have? If you don't like 8080, point me concrete number

**Comment by alex-anakin** [comment]
Now project hasn't values for `aes.init.vector.hex` and aes.key.hex` neither in *global.properties* nor in *local.properties.template*. It would be problem for new team member (for example). I propose place the above-mentioned values in global.properties. When we will start production server then we can secure these values.

**Comment by romach** [comment]
@vaa25 what about

> Now project hasn't values for `aes.init.vector.hex` and `aes.key.hex` neither in *global.properties* nor in *local.properties.template*. It would be problem for new team member (for example). I propose place the above-mentioned values in global.properties. When we will start production server then we can secure these values.

**Comment by vaa25** [comment]
@romach what what about?

File: restservice/src/main/resources/local.properties.TEMPLATE [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # Create copy of this file in same directory with name "local.properties" and update value of properties
+# There you can replace any property you need from default.properties or add new secret property
```
Comment (romach) [inline restservice/src/main/resources/local.properties.TEMPLATE]: Rename `default.properties` to `global.properties`

**Comment by romach** [comment]
> @romach what what about?

@vaa25 do we nead `aes.`-properties in `global.properties`?

**Comment by vaa25** [comment]
@romach we haven't aes in master yet

**Comment by romach** [comment]
@AndriyBaibak time spent: 30 minutes

**Comment by romach** [comment]
* **PASSED**

**Pull request merged**